.github/DEVELOPMENT.md

```diff
@@ -0,0 +1,115 @@
+
+# Environment Setup
+You will need to the protobuf compilers. In MacOs you can use brew to install it.
+```
+brew install protobuf
+```
+
+You'll need to install poetry in order to install dependencies using the lock file in this project. Follow [their docs](https://python-poetry.org/docs/) to get it set up.
+
+```
+# Use poetry to install dependencies
+make install
+
+# Build the protobuf source and the distribution tar and wheel
+make
+
+# For building and viewing the docs if you're working on those
+make docs
+```
+
+Poetry manages virtualenvs as well. Typically, on a project that uses virtualenv directly you would activate the virtualenv to get all of the binaries that you install with pip onto the path. Poetry works in a similar way but with different commands.
+
+```
+# Activate the poetry virtualenv
+poetry shell
+```
+
+You shouldn't have to actually do that unless you want to run some of the executables that are installed from pypi directly. One use case would be manually running `pytest` on a single test file rather than running them all with `make test`.
+
+## Manging Dependencies
+This is done through poetry. There needs be a good reason to add dependencies that should be included in the commit message.
+
+```bash
+# Add a dependency
+poetry add package-name
+
+# Add a dev only dependency that gets installed locally during dev
+poetry add --dev package-name
+
+# Update the lock file. Run this after adding or removing things. It doesn't appear to always  happen automatically.
+poetry lock
+```
+
+# Development Details
+You can run `make help` for a full list of targets that you can run. These are the ones that you'll need most often.
+
+```bash
+# Build everything, producing a source tar and wheel in ./dist
+make
+
+# For running tests locally
+make test
+
+# For running notebook tests
+make test-notebooks
```

Comment:
we still don't have that comment, but is TODO. maybe we should remove this mention until we have it?

Reply:
same the bump version targets, e.g. we need to bring back `bump-minor`